### PR TITLE
Haiku OS support

### PR DESCRIPTION
**Pull request by exstrim401**

### CODE REVIEW

File: engine/wscript [hunk 1]
```diff
@@ -62,6 +62,10 @@ def configure(conf):
 			conf.fatal('SDL2 not availiable! If you want to build dedicated server, specify --dedicated')
 		conf.define('XASH_SDL', 2)
 
+	if conf.env.DEST_OS == 'haiku':
+		conf.env.LIB_HAIKU = ['network']
```
Comment (a1batross) [inline engine/wscript:66]: Haiku has nothing similar to pkg-config?

Comment (a1batross) [reply engine/wscript:66]: Also is this library guaranteed to always exist at this path?

Comment (a1batross) [reply engine/wscript:66]: Otherwise lgtm.

Does port really work?

Comment (exstrim401) [reply engine/wscript:66]: network is system library needed to sockets work. it is guaranteed to always exists at this path.
port is completely working

Comment (exstrim401) [reply engine/wscript:66]: haiku has pkg-config but i'm not famillar with waf

Comment (a1batross) [reply engine/wscript:66]: Nah, pkg-config is good (there is check_pkg method for it) but it's mostly used for something that's not guaranteed to exist at certain location.

For Haiku it means we probably can't crosscompile engine but I doubt there is real necessity in it.

Comment (a1batross) [reply engine/wscript:66]: Gonna merge it now.

For future patches: please use our commit message style, it's explained in CONTRIBUTING.md.

**Pull request merged**